engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -275,9 +277,18 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
       factory <- findFactory(sourceAndResolvers._1).toIOChecked
       outputRuntimeExtractor <- factory.womOutputRuntimeExtractor.toValidated.toIOChecked
       validatedNamespace <- buildValidatedNamespace(factory, sourceAndResolvers._1, sourceAndResolvers._2)
+      closeResult = sourceAndResolvers._2.traverse(_.close())
```

Comment:
TOL: I just realized that this calls `close()` on our http importer too. I assume this is OK because the HTTP importer `close()` doesn't actually do anything, although that in itself might be confusing.

I wonder whether we should rename this method to `removeTempDirectories` to capture than nuance a little clearer?

Reply:
Does `removeTempDirectories` make sense on an HTTP resolver that has no concept of directories?

Reply:
How about `cleanupIfNecessary`